cmd/query_risks_privilegeEscalation.go

```diff
@@ -34,107 +35,117 @@ var queryRisksPrivilegeEscalationCmd = &cobra.Command{
+		accountID, err := cmd.Flags().GetString(FLAG_ACCOUNT)
+		analysisDate, err := cmd.Flags().GetString(FLAG_ANALYSIS_DATE)
+		reportHome, err := cmd.Flags().GetString(FLAG_REPORT_HOME)
+		stdout := cmd.OutOrStdout()
+		stderr := cmd.ErrOrStderr()
 
-		verbose, _ := cmd.Flags().GetBool(`verbose`)
-		customerID := cmd.Flags().Lookup(`customer_id`).Value.String()
-		accountID := cmd.Flags().Lookup(`account`).Value.String()
-		analysisDate, _ := cmd.Flags().GetString(`analysis-date`)
-
-		// load the local report database
-		db, err := core.LoadLocalDB(cmd.Flags().Lookup(`report-home`).Value.String())
 		if err != nil {
-			fmt.Printf("Unable to load local database, %v\n", err)
-		}
-		if verbose {
-			defer func() {
-				total, accounts, customers := db.Sizes()
-				fmt.Fprintf(cmd.ErrOrStderr(),
-					"Local database:\n\tCustomers:\t\t%v\n\tAccounts:\t\t%v\n\tTotal analysis dates: \t%v\n",
-					customers, accounts, total)
-			}()
+			fmt.Fprintf(stderr, err.Error())
+			os.Exit(1)
+			return
 		}
 
-		// determine the file name for the desired report
-		var path string
-		if len(analysisDate) > 0 {
-			reportDateTime, err := time.Parse(core.FILENAME_TIMESTAMP_ANALYSIS_DATE_LAYOUT, analysisDate)
-			if err != nil {
-				fmt.Fprintf(cmd.ErrOrStderr(), "Invalid analysis-date: %v\n", analysisDate)
-				os.Exit(1)
-				return
-			}
-			if qr := db.GetPathForCustomerAccountTimeKind(
-				customerID, accountID, reportDateTime,
-				core.REPORT_TYPE_PREFIX_PRINCIPALS); qr != nil {
-				path = *qr
-			} else {
-				fmt.Fprintf(cmd.ErrOrStderr(),
-					"No such report: %v, %v, %v, total records: %v\n",
-					customerID, accountID,
-					reportDateTime.Format(core.FILENAME_TIMESTAMP_ANALYSIS_DATE_LAYOUT),
-					db.Size())
-				os.Exit(1)
-				return
-			}
-		} else {
-			// latest
-			fmt.Fprintln(cmd.ErrOrStderr(), `running latest report`)
-
-			if qr := db.GetPathForCustomerAccountLatestKind(
-				customerID, accountID, core.REPORT_TYPE_PREFIX_PRINCIPALS); qr != nil {
-				path = *qr
-			} else {
-				// TODO handle no such report
-				fmt.Fprintf(cmd.ErrOrStderr(),
-					"No such report: %v, %v, total records: %v\n",
-					customerID, accountID, db.Size())
-				os.Exit(1)
-				return
-			}
-		}
+		DoQueryRisksPrivilegeEscalation(stdout, stderr, reportHome, customerID, accountID, analysisDate, format, verbose)
+	},
+}
 
-		f, err := os.Open(path)
+func init() {
+	queryRisksCmd.AddCommand(queryRisksPrivilegeEscalationCmd)
+}
+
+func DoQueryRisksPrivilegeEscalation(stdout, stderr io.Writer, reportHome, customerID, accountID, analysisDate, format string, verbose bool) {
+	// load the local report database
+	db, err := core.LoadLocalDB(reportHome)
+	if err != nil {
+		fmt.Printf("Unable to load local database, %v\n", err)
+	}
+	if verbose {
+		defer func() {
+			total, accounts, customers := db.Sizes()
+			fmt.Fprintf(stderr,
+				"Local database:\n\tCustomers:\t\t%v\n\tAccounts:\t\t%v\n\tTotal analysis dates: \t%v\n",
+				customers, accounts, total)
+		}()
```

Comment:
Could DRY-out this repeated defer with a `db.PrintStats` function.

Reply:
DRYed